### DISCUSSION TITLE
What is the current support status of WebGPU on iOS? It's originally available as an experimental feature of Safari in iOS 13. But now is missin in iOS 15.

### DISCUSSION THREAD

**Discussion by Kangz** [gpuweb/gpuweb#3245] [Q&A] [answered]
What is the current support status of WebGPU on iOS? It's originally available as an experimental feature of Safari in iOS 13. But now is missin in iOS 15.

_Originally posted by @SetoKaiba in https://github.com/gpuweb/gpuweb/issues/1069#issuecomment-1195607608_

**Comment by Kangz** [comment, marked as answer]
Apple would be the only ones that can answer this completely so take what's below with a grain of salt. @litherum feel free to correct me if I'm saying something wrong. 

AFAIK the "WebGPU" exposed by Safari was their previous "WebMetal" implementation and just one of the very initial proposals for this standard. WebGPU is currently being implemented in WebKit (which powers Safari) with some of the code being [here](https://github.com/WebKit/WebKit/tree/main/Source/WebGPU/WebGPU). There's some movement but it looks like it is implementing only a fraction of the API surface so it will take at least some time before it can be shipped on iOS.

**Reply by ghost** [reply]
It doesn't look like it's making any significant progress whatsoever. Why remove the old (partially working) implementation, when the replacement isn't ready whatsoever, nor the development of it prioritized?

To me, it seems like Apple is trying to backpedal on the whole WebGPU thing.

**Reply by ghost** [reply]
And it would also be nice if the Wiki in this repo could be updated to reflect the new lack of availability in Safari TP, since saying it's available when it isn't is misleading.
